app/util/data_preparation/jsd_prepare_data.py

```diff
@@ -0,0 +1,263 @@
+        service_desk_dict['projectId'] = service_desk['projectId']
+        service_desk_dict['projectKey'] = service_desk['projectKey']
+        service_desks_issues.append(service_desk_dict)
+
+    issues_disctribution_perc = __calculate_issues_per_project(len(service_desks))
+    issues_disctribution_id = {}
+    for key, value in issues_disctribution_perc.items():
+        issues_disctribution_id[service_desks_issues[key-1]['projectKey']] = value
+
+    print(f'Start retrieving issues by distribution per project: {issues_disctribution_id}')
+    distribution_success = True
+    issues_list = []
+    for service_desk in service_desks_issues:
+        issues = jira_api.issues_search(jql=f'project = {service_desk["projectKey"]}',
+                                        max_results=issues_disctribution_id[service_desk['projectKey']])
+
+        issues_per_project_list = [','.join((issue['id'],
+                                             issue['key'],
+                                             service_desk['serviceDeskId'],
+                                             service_desk['projectId'],
+                                             service_desk['projectKey'])) for issue in issues]
+
+        issues_list.extend(issues_per_project_list)
+        if len(issues) < issues_disctribution_id[service_desk['projectKey']]:
+            print(f'Stop retrieving issues by distribution. Project {service_desk["projectKey"]} '
+                  f'has {len(issues)}/{issues_disctribution_id[service_desk["projectKey"]]} issues')
+            distribution_success = False
+            break
+
+    if distribution_success:
+        print(f'Issues retrieving by distribution per project finished successfully. '
+              f'Retrieved {len(issues_list)} issues')
+        print('---------------------------------------------------')
+        return issues_list
+
+    print(f'Force retrieving {TOTAL_ISSUES_TO_RETRIEVE} issues from Service Desk projects')
+    projects_keys = ','.join([project['projectKey'] for project in service_desks])
+    issues = jira_api.issues_search(jql=f"project in ({projects_keys})", max_results=TOTAL_ISSUES_TO_RETRIEVE)
+    if len(issues) < TOTAL_ISSUES_TO_RETRIEVE:
+        raise Exception(f"ERROR: Jira Service Desks does not have enough issues "
+                        f"{len(issues)}/{TOTAL_ISSUES_TO_RETRIEVE}")
+    issues_list = []
+    for issue in issues:
+        for service_desk in service_desks_issues:
+            if service_desk['projectKey'] in issue['key']:
+                issue_str = f'{issue["id"]},' \
+                            f'{issue["key"]},' \
+                            f'{service_desk["serviceDeskId"]},' \
+                            f'{service_desk["projectId"]},' \
+                            f'{service_desk["projectKey"]}'
+
+                issues_list.append(issue_str)
+    print(f"Force retrieved {len(issues_list)} issues.")
+    print('---------------------------------------------------')
```

Comment:
This print is required or it was for your convenience?

Reply:
This print is a line to dedicate output from prepare_data.py. It is not required, we can freely delete it after all test.